google_analytics/preprocessing.py

```diff
@@ -111,3 +124,95 @@ def keep_intersection_of_columns(train, test):
     """
     shared_cols = list(set(train.columns).intersection(set(test.columns)))
     return train[shared_cols], test[shared_cols]
+
+
+def replace(string, regex, replacement):
+    """ replace an original string when a regular expression is in this string
+
+    params
+    ------
+    string: The original string
+    regex: Regular expression
+    replacement: The replacing string
+
+    return
+    ------
+    When the regular expression is in the original string, the replacement is returned.
+    Else the original string is returns
+    """
+    regex = re.compile(regex)
+    if pd.isnull(string):
+        return string
+    else:
+        if regex.search(string):
+            return replacement
+        else:
+            return string
+
+
+def isSimilar(string, aimed_string, threshold):
+    """ Indicates whether string is similar to aimed_string including the levenshtein distance
+
+    params
+    ------
+    string: The original string
+    aimed_string: The aimed string
+    threshold: The maximum levenshtein distance
+
+    return
+    ------
+    label: boolean
+        Is True when the levenshtein distance between string and aimed_string
+        is smaller or equal to the threshold. Else it returns False.
+    score: int
+        The levenshtein distance between string and aimed_string
+    """
+    if pd.isnull(string):
+        return np.nan, np.nan
+    else:
+        score = Levenshtein.distance(string.lower(), aimed_string)
+        return_score = np.nan
+        label = False
+        if score > threshold:
+            for word in string.split():
+                score = Levenshtein.distance(word.lower(), aimed_string)
+                if score <= threshold:
+                    label = True
+                    return_score = score
+                    break
+        else:
+            label = True
+            return_score = score
+        return label, return_score
+
+
+def give_cat(string):
```

Comment:
It will be helfpul, for future use., to set two pamateres:  The original source and the category. Here you have predefined the 5 target categories. But you can make it more general.  I am thinking something like 
`def give_cat(string, category): 
regex_cat = re.compile(category)
if regex_google.search(string): return category else return np.nan`